### PR TITLE
fix: Final harvest does not happen when initial RUM call fails

### PR DESCRIPTION
**Pull request by cwli24**
Fixes the final harvest behavior in cases when the RUM request fails. The initial request is meant to act as a gateway for the agent to start reporting data, and its failure should result in the agent not sending anything. As of previous version, the metrics and page view timing features do not follow this rule, and that has been corrected.
---

### Overview

This cuts down on disassociated data from user page "sessions" wherein their agent never sent any RUM or interval analytics yet does sent disjoint metrics and timings that cannot be tied to anything else of said page session.

### Related Issue(s)

NR-138551

### Testing

Tests added but can also manually verify that following a failed RUM request, nothing else gets send either on interval or on refresh (sendBeacon).

### CODE REVIEW

**Review by patrickhousley — APPROVED** [review]

File: src/features/metrics/aggregate/index.js
Comment (metal-messiah) [inline src/features/metrics/aggregate/index.js:1]: This brings up a good point, but IMO it'd probably be simpler and more reliable if this feature just waited for the `err` flag before setting up the harvester and/or running the `checks()`, like the other features do, and we remove the whole block-err behavior.

Comment (cwli24) [reply src/features/metrics/aggregate/index.js:1]: It could but it wouldn't fix it for PVT as that doesn't depend on any flag.

Comment (metal-messiah) [reply src/features/metrics/aggregate/index.js:1]: Where in this PR is the problem being addressed for PVT?

Comment (cwli24) [reply src/features/metrics/aggregate/index.js:1]: PVT `startTimer` is tied to the drain event, but it's scheduler initializing is not. It's not just PVT as this requires all features to have "started" before being allowed to final harvest.

**Review by metal-messiah — APPROVED** [review]

**Pull request merged**